src/Foundation/NSUrlSessionHandler.cs

```diff
@@ -1083,6 +1089,16 @@ void DidReceiveChallengeImpl (NSUrlSession session, NSUrlSessionTask task, NSUrl
 					}
 					return;
 				}
+				if (sessionHandler.ClientCertificateOptions == ClientCertificateOption.Manual && challenge.ProtectionSpace.AuthenticationMethod == NSUrlProtectionSpace.AuthenticationMethodClientCertificate) {
+					var certificate = CertificateHelper.GetEligibleClientCertificate (sessionHandler.ClientCertificates);
+					if (certificate is not null) {
+						var cert = new SecCertificate (certificate);
+						var identity = SecIdentity.Import (certificate);
+						var credential = new NSUrlCredential (identity, new SecCertificate [] { cert }, NSUrlCredentialPersistence.ForSession);
+						completionHandler (NSUrlSessionAuthChallengeDisposition.UseCredential, credential);
+						return;
+					}
+				}
```

Comment:
This follows how the other Http Handlers handles multiple certificates specified. Could you provide a repro that shows how intermediate certificates are needed for PKI authentication and how that works in for instance SocketsHttpHandler? That would be helpful to verify whether such a change is really needed. In my testing I didn't see this was needed for user authentication

Reply:
As @wfurt points out [here](https://github.com/dotnet/runtime/issues/47680#issuecomment-771093045):

> The handler.ClientCertificates is really meant for the leaf certificates

Reply:
I don't have time to create a complex repro. sadly, but it may not be needed. Consider the scenario where the server wants to authenticate users from various places with various intermediate CAs in the certificate chain. This PR only passes in the end-entity certificate, which may fail to validate if the server doesn't have every possible intermediate CA certificate installed. (Certain government agencies, for example, do not use OS-trusted CAs.)

This is why [`NSURLCredential`](https://developer.apple.com/documentation/foundation/nsurlcredential/1428192-credentialwithidentity) takes an array of intermediate certificates to begin with.

This PR also appears to populate the `certArray` with non-intermediate certificates, which is also probably incorrect.

Reply:
> As @wfurt points out [here](https://github.com/dotnet/runtime/issues/47680#issuecomment-771093045):
> 
> > The handler.ClientCertificates is really meant for the leaf certificates

OK. If we want to go that route and align the macOS behavior to quirky Windows behavior, we should minimally pass in all intermediates to `NSUrlCredential` (e.g., whatever is in `StoreName.CertificateAuthority`, replacing `new SecCertificate [] { cert }`)